reana_client/cli/workflow.py

```diff
@@ -136,21 +142,26 @@ def workflow_workflows(ctx, sessions, _filter, output_format, access_token,
             name, run_number = get_workflow_name_and_run_number(
                 workflow['name'])
             workflow['name'] = name
-            workflow['run_number'] = run_number
+            workflow['run_number'] = int(run_number)
             if type == 'interactive':
                 workflow['session_uri'] = format_session_uri(
                     reana_server_url=ctx.obj.reana_server_url,
                     path=workflow['session_uri'],
                     access_token=access_token)
-            data.append([str(workflow[k]) for k in headers[type]])
-        data = sorted(data, key=lambda x: int(x[1]))
+            data.append([workflow[k] for k in headers[type]])
+        sort_column_id = 2
+        if sort_columm_name in headers[type]:
```

Comment:
```suggestion
        if sort_columm_name.lower() in headers[type]:
```
And ordering for the other columns will work :)